File: Guanabara - POO Java/Meus exercícios POO/Aula12.Polimorfismo/src/main/java/polimorfismo/Peixe.java
package polimorfismo;
public class Peixe extends Animal { //tipo de herança: 'para diferença'
    //Atributo
    private String corEscama;
    
    //Método Público
    public void soltarBolha() {
        System.out.println("Soltando bolhas...");
    }
    
    //Implementando Métodos Abstratos (Polimorfismo de Sobreposição)
    @Override
    public void locomover() {
        System.out.println("Nadando...");
    }

    @Override
    public void alimentar() {
        System.out.println("Comendo substâncias...");
    }

    @Override
    public void emitirSom() {
        System.out.println("Peixe não faz som!");
    }
    
    //Métodos Acessores
    public String getCorEscama() {
        return corEscama;
    }

    public void setCorEscama(String corEscama) {
        this.corEscama = corEscama;
    }
}

File: Guanabara - PHP/exercicios-PHPOO/Aula12-Polimorfismo/Peixe.php
<?php
include_once 'Animal.php';
class Peixe extends Animal
{
    //Atributos
    protected $corEscama;
    
    //Métodos Especiais
    public function soltarBolha(){
        echo "<p>Peixe soltando bolhas...</p>";
    }
    
    //Métodos Implementados usando polimorfismo de sobreposição
    public function alimentar() {
        echo '<p>Peixe se alimentando...</p>';
    }
    public function emitirSom() {
        echo '<p>Peixe não emite som!</p>';
    }
    public function locomover() {
        echo '<p>Peixe nadando...</p>';
    }
    
    //Métodos Acessores
    public function getCorEscama() {
        return $this->corEscama;
    }

    public function setCorEscama($corEscama): void {
        $this->corEscama = $corEscama;
    }
}
